File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = -float('inf')
        current_sum = 0
        for num in nums:
            if current_sum + num < num:
                current_sum = num
            else:
                current_sum += num
            
            max_sum = max(max_sum, current_sum)
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int current_max = 0;
        int max_number = Integer.MIN_VALUE;
        for(int num : nums){
            current_max = Math.max(current_max + num, num);
            max_number = Math.max(max_number, current_max);
        }
        return max_number;
    }
}